### PR TITLE
Implement reaction picker and reaction list

### PR DESCRIPTION
**Pull request by hochiw**
## Testing
- Mutation changes query result
- Tested reaction flow (set reaction, unset reaction)
    - Set Reaction
    - Unset Reaction
- Tested logged out flow
    - Logged out, Set Reaction
    - Logged out, Unset Reaction
- Tested wallet disconnected reaction flow, handle in another issue oursky/likedao#178 

## Pending
- Active reaction badge coloring

## Screenshots
| Desktop | Mobile |
|---------|--------|
|Screenshot 2022-06-27 at 4 21 47 PM|Screenshot 2022-06-27 at 4 21 54 PM|
|Screenshot 2022-06-27 at 4 22 02 PM|Screenshot 2022-06-27 at 4 22 09 PM|

refs oursky/likedao#116

### CODE REVIEW

**Comment by hochiw** [comment]
wait for #156 to merge before rebase

**Comment by chihimng** [comment]
Plz add screenshot? 🙏

File: react-app/src/providers/AppApolloProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  introspection as unknown as IntrospectionQuery
+);
+
+interface AppApolloProviderProps {
+  children?: React.ReactNode;
+}
+
+const AppApolloProvider: React.FC<AppApolloProviderProps> = (props) => {
+  const { children } = props;
+  const [isLogginIn, setIsLogginIn] = useState(false);
```
Comment (chihimng) [inline react-app/src/providers/AppApolloProvider.tsx:67]: typo?

File: react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts [hunk 2]
```diff
@@ -26,6 +32,52 @@ const calculateTurnout = (tallyResult: Proposal["tallyResult"]) => {
...
+    ProposalDetailSetReactionMutationMutation,
+    ProposalDetailSetReactionMutationMutationVariables
+  >(ProposalDetailSetReactionMutation);
+
+  return useCallback(
+    async (proposalId: string, reaction: ReactionType) => {
+      return mutationFunction({
+        variables: {
+          input: {
+            targetId: `proposal_${proposalId}`,
```
Comment (chihimng) [inline react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts:49]: client not supposed to format node ids... any easy way to accept node id instead?

File: react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts [hunk 3]
```diff
@@ -26,6 +32,52 @@ const calculateTurnout = (tallyResult: Proposal["tallyResult"]) => {
   return turnout.toNumber();
 };
 
+export function useSetProposalReactionMutation(): (
```
Comment (chihimng) [inline react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts:35]: can we have a generic set reaction api that accepts node id as input? & reuse it across whole app?

File: react-app/src/components/ProposalDetailScreen/ProposalDetailScreen.tsx [hunk 4]
```diff
@@ -11,15 +11,47 @@ import {
...
 
 const ProposalDetailScreen: React.FC = () => {
   const { id } = useParams();
   const { fetch, requestState } = useProposalQuery();
+  const { translate } = useLocale();
+  const setProposalReaction = useSetProposalReactionMutation();
+  const unsetProposalReaciton = useUnsetProposalReactionMutation();
   const navigate = useNavigate();
 
+  const onSetReaction = useCallback(
```
Comment (chihimng) [inline react-app/src/components/ProposalDetailScreen/ProposalDetailScreen.tsx:32]: can we refactor this to be reused for other reaction targets in the future?

**Review by elise-ng — CHANGES_REQUESTED** [review]
& plz help describe what was tested? like login on first auth error, and does the apollo cache work to reflect new action after mutation?

File: react-app/src/components/reactions/ReactionPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  });
+
+  const reactions = useMemo(
+    (): Reaction[] =>
+      Object.values(DefaultReactionMap).map((r) => {
+        if (r.label === ReactionType.Likecoin) {
+          return {
+            label: r.label,
+            node: (
+              <span className={cn("text-3xl", "leading-none")}>{r.node}</span>
```
[the remark this answers is not in the document]
Comment (hochiw) [reply react-app/src/components/reactions/ReactionPicker.tsx:41]: cos the size for the likecoin icon is different for the list and the picker

[the remark this answers is not in the document]
Comment (hochiw) [reply react-app/src/components/reactions/ReactionPicker.tsx:41]: unless i separate it into `pickerNode` and `listNode`

[the remark this answers is not in the document]
Comment (hochiw) [reply react-app/src/components/reactions/ReactionPicker.tsx]: resolved with css

**Comment by hochiw** [comment]
@chihimng updated 🙏

**Review by elise-ng — APPROVED** [review]

**Pull request merged**